batch/batch/driver/main.py

```diff
@@ -1014,13 +1015,21 @@ async def check(tx):
     CAST(COALESCE(SUM(state = 'Creating' AND cancelled), 0) AS SIGNED) AS actual_n_cancelled_creating_jobs
   FROM
   (
-    SELECT batches.user, jobs.state, jobs.cores_mcpu, jobs.inst_coll,
-      (jobs.always_run OR NOT (jobs.cancelled OR job_groups_cancelled.id IS NOT NULL)) AS runnable,
-      (NOT jobs.always_run AND (jobs.cancelled OR job_groups_cancelled.id IS NOT NULL)) AS cancelled
-    FROM batches
-    INNER JOIN jobs ON batches.id = jobs.batch_id
-    LEFT JOIN job_groups_cancelled ON batches.id = job_groups_cancelled.id
-    WHERE batches.`state` = 'running'
+    SELECT job_groups.user, jobs.state, jobs.cores_mcpu, jobs.inst_coll,
```

Comment:
Continuing from https://github.com/hail-is/hail/pull/14170/files#r1476851646

Hmm. OK. Including the user lets us partition the groups table between users and quickly find a single users running groups. That does seem better than scanning all the running batches. There could be many more groups which are not running than there are running groups. It is unfortunate that this knowledge isn't represented in the code anywhere, but I don't see an obvious way to encode this in code. It seems best as a component of a dev doc on scheduling.